### DISCUSSION TITLE
Provisioning of Flux on Workload Clusters

### DISCUSSION THREAD

**Discussion by lieberlois** [telekom/das-schiff#20] [Q&A] [answered]
I love this setup, and did a very similar thing in my master thesis. I did not quite understand, how workload clusters are provisioned with FluxCD initially (classic chicken-egg problem). I solved it by using a HelmChartProxy ([https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/blob/main/docs/quick-start.md#4-example-install-nginx-ingress-to-the-workload-cluster](exmaple)) but this doesn't seem to be the case in this project, since the flux manifests are stored in the cluster-components folder of the according workload cluster.

Same question for secrets (e.g. Git Tokens etc.) ^^

Any ideas? Thanks in advance, and again: love the setup! :)
Cheers Luis

**Comment by schrej** [comment, marked as answer]
We're applying flux and network plugins from our management clusters using Kustomizations that reference the tenant clusters' Kubeconfig ConfigMaps. This also includes secrets. We have rather complex ignore files to filter what exactly will be applied.
